### DISCUSSION TITLE
LCM model generates only noise

### DISCUSSION THREAD

**Discussion by Arctomachine** [vladmandic/sdnext#2471] [Q&A]
I have freshly installed this client and downloaded lcm dreamshaper. But it only generates same noise it had initially with any number of steps, with any sampler, at any seed, at any cfg scale - even if following recommended settings from description 1-to-1.
With vae it generates noise, but without vae it just outputs plain grey square.
Where could the problem be? Am I missing some workflow steps?
I am using sub quadratic attention mode and --lowvram flag, the rest of system settings should be default

**Comment by vladmandic** [comment]
i have to ask a trivial question - how did you download a model? i'm asking because the rest you say is directly opposite to announcement of support for LCM which explicitly state that only sampler=default is supported since model comes with its own. and you cannot use generic VAE on something that is clearly not an SD model. i mean, LCM is NOT SD. you cannot use SD schedulers and you cannot use SD VAE.

**Reply by Arctomachine** [reply]
Downloaded it from huggingface, then it pulled metadata from civitai, where description says to use dpm++ 2m karras.
I am trying now to load model with default sampler enabled and with vae set to none, but it gives black square with other settings set as default.
Also there seems to be no way to select default sampler again until full interface reload once it has been changed to something else.

**Reply by vladmandic** [reply]
notes clear state: 
> select from extra networks -> reference or download using built-in Huggingface downloader: SimianLuo/LCM_Dreamshaper_v7

you downloading model manually makes app not know its LCM at all, its trying to load it as SD model and guess what? it doesn't work.

**Reply by Arctomachine** [reply]
I deleted old model and downloaded again from models/huggingface tab. It said downloaded successfully, but model itself is not appearing anywhere in interface even after restarting server and client. Where is it?

**Reply by vladmandic** [reply]
you need to be running in backend:diffusers and it shows up under networks -> models -> diffusers (and networks -> models -> reference)
if you're running in backend:original, you only have access to sd15/sd21. backend:diffusers is required for ALL other models. that is clearly documented in readme.

**Reply by Arctomachine** [reply]
Switched to diffusers backend and selected xformers mode, but now it gives error when trying to run generation
```
AttributeError: 'LatentConsistencyModelPipeline' object has no attribute 'model'
19:50:08-079245 ERROR    Exception: 'NoneType' object is not callable
```
What else am I missing?

**Reply by vladmandic** [reply]
based on a single line without context? no idea what you're missing since i really don't know how you got there.
but it looks like model is not loaded.

**Reply by Arctomachine** [reply]
If I start from very beginning, what are correct steps to make it work on clean install?

**Reply by vladmandic** [reply]
1. start sdnext + switch backend to diffusers + restart sdnext  
  or start in diffusers mode directly using `webui --backend diffusers`
2. open networks -> models -> reference
3. select lcm model  
  it will get automatically downloaded (to models/Diffusers) if not already on the system and loaded.

**Comment by gsgoldma** [comment]
I don't even get noise

01:14:32-175503 INFO     Applying free-u: b1=1.2 b2=1.4 s1=0.9 s2=0.2
01:14:41-702915 INFO     Applying hypertile: unet=256
01:14:41-723914 ERROR    Exception: 'NoneType' object is not callable
01:14:41-726915 ERROR    Arguments: args=('task(c9wbb7mt6ytcjdz)', 'uiyui', '', [], 18, None, None, False, False,
                         False, 1, 1, 6, 6, 0.7, 1, -1.0, -1.0, 0, 0, 0, 512, 512, False, 0.5, 2, 'None', False, 20, 0,
                         0, 5, 0.8, '', '', [], 0, False, '', True, False, 0, 384, 0, True, True, True, 1, '\n
                         <h3><strong>Combinations</strong></h3>\n            Choose a number of terms from a list, in
                         this case we choose two artists\n            <code>{2$$artist1|artist2|artist3}</code>\n
                         If $$ is not provided, then 1$$ is assumed.\n            <br/><br/>\n\n
                         <h3><strong>Wildcards</strong></h3>\n            <p>Available wildcards</p>\n
                         <ul>\n        </ul>\n            <br/>\n            <code>WILDCARD_DIR:
                         scripts/wildcards</code><br/>\n            <small>You can add more wildcards by creating a
                         text file with one term per line and name is mywildcards.txt. Place it in scripts/wildcards.
                         <code>__mywildcards__</code> will then become available.</small>\n        ', None, '',
                         'outputs', 1, '', 0, '', True, False, False, False, False, False, 1, 1, False, True, False,
                         '', 1, True, 100, True, 50, 8, True, 16, 'Median cut', 8, True, True, 16, 'PNN', True, False,
                         None, None, '', '', '', '', 'Auto rename', {'label': 'Upload avatars config'}, 'Open outputs
                         directory', 'Export to WebUI style', True, {'label': 'Presets'}, {'label': 'QC preview'}, '',
                         [], 'Select', 'QC scan', 'Show pics', None, False, False, 'positive', 'comma', 0, '', 25,
                         True, 5.0, False, False, '', '', '', '', 0.0, 'mizuni-concept-v1', 1.0, 0, 'Median cut',
                         'luminance', False, 'svg', True, True, False, 0.5, 0, '', [], 0, '', [], 0, '', [], False,
                         True, False, False, False, False, 0, False, 4.0, '', 10.0, 'Linear', 3, False, 30.0, True,
                         False, False, 0, 0.0, 'Lanczos', 1, True, 0, 0, 0.001, 75, 0.0, False, True, False, 7, 100,
                         'Constant', 0, 'Constant', 0, 4, True, 'MEAN', 'AD', 1, False, 'x264', 'blend', 10, 0, 0,
                         False, True, True, True, 'intermediate', 'animation') kwargs={}
01:14:41-799916 ERROR    gradio call: TypeError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\automatic\modules\call_queue.py:34 in f                                                                          │
│                                                                                                                     │
│   33 │   │   │   try:                                                                                               │
│ > 34 │   │   │   │   res = func(*args, **kwargs)                                                                    │
│   35 │   │   │   │   progress.record_results(id_task, res)                                                          │
│                                                                                                                     │
│ D:\automatic\modules\txt2img.py:66 in txt2img                                                                       │
│                                                                                                                     │
│   65 │   if processed is None:                                                                                      │
│ > 66 │   │   processed = processing.process_images(p)                                                               │
│   67 │   p.close()                                                                                                  │
│                                                                                                                     │
│ D:\automatic\modules\processing.py:738 in process_images                                                            │
│                                                                                                                     │
│    737 │   │   │   with context_hypertile_vae(p), context_hypertile_unet(p):                                        │
│ >  738 │   │   │   │   res = process_images_inner(p)                                                                │
│    739 │   finally:                                                                                                 │
│                                                                                                                     │
│ D:\automatic\modules\processing.py:891 in process_images_inner                                                      │
│                                                                                                                     │
│    890 │   │   │   │   from modules.processing_diffusers import process_diffusers                                   │
│ >  891 │   │   │   │   x_samples_ddim = process_diffusers(p, p.seeds, p.prompts, p.negative_pro                     │
│    892 │   │   │   else:                                                                                            │
│                                                                                                                     │
│ D:\automatic\modules\processing_diffusers.py:432 in process_diffusers                                               │
│                                                                                                                     │
│   431 │   try:                                                                                                      │
│ > 432 │   │   output = shared.sd_model(**base_args) # pylint: disable=not-callable                                  │
│   433 │   except AssertionError as e:                                                                               │
│                                                                                                                     │
│                                              ... 14 frames hidden ...                                               │
│                                                                                                                     │
│ D:\automatic\venv\lib\site-packages\torch\nn\modules\module.py:1527 in _call_impl                                   │
│                                                                                                                     │
│   1526 │   │   │   │   or _global_forward_hooks or _global_forward_pre_hooks):                                      │
│ > 1527 │   │   │   return forward_call(*args, **kwargs)                                                             │
│   1528                                                                                                              │
│                                                                                                                     │
│ D:\automatic\venv\lib\site-packages\transformers\models\clip\modeling_clip.py:385 in forward                        │
│                                                                                                                     │
│    384 │   │                                                                                                        │
│ >  385 │   │   hidden_states = self.layer_norm1(hidden_states)                                                      │
│    386 │   │   hidden_states, attn_weights = self.self_attn(                                                        │
│                                                                                                                     │
│ D:\automatic\venv\lib\site-packages\torch\nn\modules\module.py:1518 in _wrapped_call_impl                           │
│                                                                                                                     │
│   1517 │   │   else:                                                                                                │
│ > 1518 │   │   │   return self._call_impl(*args, **kwargs)                                                          │
│   1519                                                                                                              │
│                                                                                                                     │
│ D:\automatic\venv\lib\site-packages\torch\nn\modules\module.py:1527 in _call_impl                                   │
│                                                                                                                     │
│   1526 │   │   │   │   or _global_forward_hooks or _global_forward_pre_hooks):                                      │
│ > 1527 │   │   │   return forward_call(*args, **kwargs)                                                             │
│   1528                                                                                                              │
│                                                                                                                     │
│ D:\automatic\extensions-builtin\Lora\networks.py:355 in network_LayerNorm_forward                                   │
│                                                                                                                     │
│   354 │   network_apply_weights(self)                                                                               │
│ > 355 │   return originals.LayerNorm_forward(self, input)                                                           │
│   356                                                                                                               │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
TypeError: 'NoneType' object is not callable

**Reply by vladmandic** [reply]
@gsgoldma i have no idea what you're doing, dumping log in a bad format without any description of the workflow is not going to help.

**Reply by gsgoldma** [reply]
Damn. I'll prove more info. in a separate issue log. I apologize.
